client/src/redux/fetch-user-saga.js

```diff
@@ -9,8 +9,22 @@ import {
 
 function* fetchSessionUser() {
   try {
-    const { data: user } = yield call(getSessionUser);
+    const { res, timeout } = yield race({
+      res: call(getSessionUser),
+      timeout: delay(5000)
```

Comment:
The only thing to note is it is entirely possible for the request to succeed, but take longer than 5s. In which case, the client ignores the success.

So, I suggest also aborting the fetch after 5s. _According to the Google, Chrome's fetch times out after 300s._

You should be able to remove all this timeout stuff here in place of an abort signal on the fetch...

Reply:
So, skip the race entirely and instead set a timeout to abort the fetch via an AbortController?

Reply:
In my opinion, that is the better approach, yes.